doc/components/popup.md

```diff
@@ -0,0 +1,481 @@
+# Popup
+
+- Spawns a buffer inside a popup
+- You can use `vim.api` to manipulate it like any other buffer
+
+## Properties & Usage
+
+```lua
+local Popup = require("nui.popup")
+
+local popup = Popup({
+    position = "20%",
+    size = {
+        width = 20,
+        height = 10,
+    },
+    enter = true,
+    focusable = true,
+    zindex = 10,
+    relative = "editor",
+    border = {
+        highlight = "MyHighlightGroup",
+        padding = {
+            top = 2,
+            bottom = 2,
+            left = 3,
+            right = 3,
+        },
+        style = "double",
+        text = {
+            top = " I am top title ",
+            top_align = "center",
+            bottom = "I am bottom title",
+            bottom_align = "left",
+        },
+    },
+    buf_options = {
+        modifiable = true,
+        readonly = false,
+    },
+    win_options = {
+        winblend = 10,
+        winhighlight = "Normal:Normal",
+    },
+)
+```
+
+## Property Descriptions
+
+### `border`
+
+- **Type:** `table`
+- **Description:** Contains all border related options
```

Comment:
🤔 Umm... This format seems a bit repetitive. How do you feel about changing it a little bit like this:


```md
**Type**: `table`

Contains all border related options
```

Reply:
Makes sense, I was trying a lot of iterations and this one seemed kinda okay but I get your point.